ros1_bridge/__init__.py

```diff
@@ -538,24 +538,27 @@ def determine_common_services(ros1_srvs, ros2_srvs, mapping_rules):
                 break
             for i, ros1_field in enumerate(ros1_fields[direction]):
                 ros1_type = ros1_field[0]
-                ros2_type = str(ros2_fields[direction][i].type)
+                ros2_type = ros2_fields[direction][i].type
                 ros1_name = ros1_field[1]
                 ros2_name = ros2_fields[direction][i].name
-                if ros1_type != ros2_type or ros1_name != ros2_name:
+                if ros1_type != str(ros2_type) or ros1_name != ros2_name:
                     match = False
                     break
+                ros2_cpptype = ros2_type.pkg_name + '::msg::' + ros2_type.type if ros2_type.pkg_name else ''
                 output[direction].append({
-                    'basic': False if '/' in ros1_type else True,
-                    'array': True if '[]' in ros1_type else False,
+                    'basic': not ros2_type.pkg_name,
+                    'array': ros2_type.is_array,
+                    'dynamic_array': ros2_type.is_array and not ros2_type.array_size and not ros2_type.is_upper_bound,
```

Comment:
Based on the above changes to the template I assume you want `dynamic_array` to be `True` for bounded as well as unbounded arrays. Then the condition needs fixing:

`ros2_type.is_array and (not ros2_type.array_size or ros2_type.is_upper_bound)`